genkit-tools/common/tests/eval/localFileDatasetStore_test.ts

```diff
@@ -0,0 +1,341 @@
+      input: 'Cats are evil',
+      reference: 'Sorry no reference',
+    },
+    {
+      input: 'Dogs are beautiful',
+    },
+  ],
+};
+
+const SAMPLE_DATASET_1_V2 = {
+  samples: [
+    {
+      input: 'Cats are evil',
+      reference: 'Sorry no reference',
+    },
+    {
+      input: 'Dogs are angels',
+    },
+    {
+      input: 'Dogs are also super cute',
+    },
+  ],
+};
+
+const SAMPLE_DATASET_ID_1 = '12345678';
+const SAMPLE_DATASET_NAME_1 = 'dataset-1';
+
+const SAMPLE_DATASET_METADATA_1_V1 = {
+  datasetId: SAMPLE_DATASET_ID_1,
+  size: 2,
+  version: 1,
+  displayName: SAMPLE_DATASET_NAME_1,
+  createTime: FAKE_TIME.toString(),
+  updateTime: FAKE_TIME.toString(),
+};
+const SAMPLE_DATASET_METADATA_1_V2 = {
+  datasetId: SAMPLE_DATASET_ID_1,
+  size: 3,
+  version: 2,
+  displayName: SAMPLE_DATASET_NAME_1,
+  createTime: FAKE_TIME.toString(),
+  updateTime: FAKE_TIME.toString(),
+};
+
+const CREATE_DATASET_REQUEST = CreateDatasetRequestSchema.parse({
+  data: SAMPLE_DATASET_1_V1,
+  displayName: SAMPLE_DATASET_NAME_1,
+});
+
+const UPDATE_DATASET_REQUEST = UpdateDatasetRequestSchema.parse({
+  patch: SAMPLE_DATASET_1_V2,
+  datasetId: SAMPLE_DATASET_ID_1,
+  displayName: SAMPLE_DATASET_NAME_1,
+});
+
+const UPDATE_DATASET_REQUEST_NO_ID = UpdateDatasetRequestSchema.parse({
+  patch: SAMPLE_DATASET_1_V1,
+  displayName: SAMPLE_DATASET_NAME_1,
+});
+
+const SAMPLE_DATASET_ID_2 = '22345678';
+const SAMPLE_DATASET_NAME_2 = 'dataset-2';
+
+const SAMPLE_DATASET_METADATA_2 = {
+  datasetId: SAMPLE_DATASET_ID_2,
+  size: 5,
+  version: 1,
+  displayName: SAMPLE_DATASET_NAME_2,
+  createTime: FAKE_TIME.toString(),
+  updateTime: FAKE_TIME.toString(),
+};
+
+jest.mock('crypto', () => {
+  return {
+    createHash: jest.fn().mockReturnThis(),
+    update: jest.fn().mockReturnThis(),
+    digest: jest.fn(() => 'store-root'),
+  };
+});
+
+jest.mock('uuid', () => ({
+  v4: () => SAMPLE_DATASET_ID_1,
+}));
+
+jest.useFakeTimers({ advanceTimers: true });
+jest.setSystemTime(FAKE_TIME);
+
+describe('localFileDatasetStore', () => {
+  let DatasetStore: DatasetStore;
+
+  beforeEach(() => {
+    LocalFileDatasetStore.reset();
+    DatasetStore = LocalFileDatasetStore.getDatasetStore() as DatasetStore;
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('createDataset', () => {
+    it('writes and updates index for new dataset', async () => {
+      fs.promises.writeFile = jest.fn(async () => Promise.resolve(undefined));
+      fs.promises.appendFile = jest.fn(async () => Promise.resolve(undefined));
+      // For index file reads
+      fs.promises.readFile = jest.fn(async () =>
+        Promise.resolve(JSON.stringify({}) as any)
+      );
+
+      const datasetMetadata = await DatasetStore.createDataset(
+        CREATE_DATASET_REQUEST
+      );
+
+      expect(fs.promises.writeFile).toHaveBeenCalledTimes(2);
+      expect(fs.promises.writeFile).toHaveBeenNthCalledWith(
+        1,
+        expect.stringMatching(
```

Comment:
How about `expect.stringContaining`? No need for a regex check IMO

Reply:
Ah, yes, that works.